devops/resources/replace-usernames.sh

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
+
+set -ex
+
+# Display the environment variables again, this time within the context of a
+# new subshell inside of the job. N.B. this should not print plain credentials
+# because the only credentialsBindings we currently use is of type "file" which
+# just stores a filename in the environment (rather than the content).
+env
+
+# Make sure that when we try to write unicode to the console, it
+# correctly encodes to UTF-8 rather than exiting with a UnicodeEncode
+# error.
+export PYTHONIOENCODING=UTF-8
+export LC_CTYPE=en_US.UTF-8
+
+mkdir -p $WORKSPACE/username-replacement-secure
+
+# prepare tubular
+cd $WORKSPACE/tubular
+pip install -r requirements.txt
+
+# Call the script to replace the usernames for all users in the CSV file.
+python scripts/replace_usernames.py \
+    --config_file=$WORKSPACE/username-replacement-secure/$ENVIRONMENT.yml # TODO: DEVOPS DROPS FILE HERE?
```

Comment:
Rather than doing this, you'll want to do something like this: https://github.com/edx/jenkins-job-dsl/blob/master/devops/jobs/CheckSesLimits.groovy#L45-L46

In that example a file that we upload to jenkins will be available at a random path and the location of the path will be available as the environment variable `AWS_CONFIG_FILE`  You'll want to do something similar here.

Reply:
Changed to match that pattern. If there is to be a separate config file for each environment, like I had with `$ENVIRONMENT.yml`, do I need to create two `credentialsBinding` => `file` objects, or can a different file be at that location for each job. I.e. is the new `tools-edx-jenkins-username-replacement-config` global or per job?

Reply:
See new comment, these names are global so you should have a different one pre environment.